src/mds/Migrator.cc

```diff
@@ -1779,7 +1794,16 @@ uint64_t Migrator::encode_export_dir(bufferlist& exportbl,
     exportbl.append("I", 1);    // inode dentry
     
     encode_export_inode(in, exportbl, exported_client_map, exported_client_metadata_map);  // encode, and (update state for) export
-    
+    if (in->batch_ops.size()) {
+      for (map<int, set<BatchOpContext*>>::iterator iter = in->batch_ops.begin();
+	  iter != in->batch_ops.end();
+	  iter++) {
+	if (iter->second.size()) {
+	  dir->subtree_inodes_with_batchops.insert(in);
+	}
+      }
+    }
+
```

Comment:
This is complex. It makes me think it's better to attach batch requests to the first request.  when finishing the first request, also finish the batch request

Reply:
Oh, yes. I'll try to modify it